jans-cedarling/cedarling/src/jwt/decoding_strategy.rs

```diff
@@ -0,0 +1,161 @@
+/*
+ * This software is available under the Apache-2.0 license.
+ * See https://www.apache.org/licenses/LICENSE-2.0.txt for full text.
+ *
+ * Copyright (c) 2024, Gluu, Inc.
+ */
+
+use super::traits::{Decode, GetKey};
+use super::Error;
+use di::DependencySupplier;
+use jsonwebtoken as jwt;
+use serde::de::DeserializeOwned;
+use std::sync::Arc;
+
+/// Represents the decoding strategy for JWT tokens.
+///
+/// This enum determines how JWT tokens are decoded, either without validation
+/// or with validation using specified algorithms and a key service. The
+/// appropriate strategy can be selected based on the bootstrap config.
+pub enum DecodingStrategy {
+    /// Decoding strategy without validation.
+    WithoutValidation,
+
+    /// Decoding strategy with validation.
+    WithValidation {
+        key_service: Arc<dyn GetKey>,
+        supported_algs: Vec<jwt::Algorithm>,
+    },
+}
+
+impl DecodingStrategy {
+    /// Creates a new decoding strategy that does not perform validation.
+    ///
+    /// This method initializes the `WithoutValidation` strategy, which
+    /// decodes JWT tokens without any validation checks
+    pub fn new_without_validation() -> Self {
+        Self::WithoutValidation
+    }
+
+    /// Decodes a JWT without performing validation.
+    fn decode_jwt_without_validation<T: DeserializeOwned>(jwt: &str) -> Result<T, Error> {
+        let header = jwt::decode_header(jwt).map_err(|e| Error::ParsingError(e))?;
+
+        let mut validator = jwt::Validation::new(header.alg);
+        validator.insecure_disable_signature_validation();
+        validator.validate_exp = false;
+        validator.validate_aud = false;
+        validator.validate_nbf = false;
+
+        let key = jwt::DecodingKey::from_secret("secret".as_ref());
+
+        let claims = jwt::decode::<T>(&jwt, &key, &validator)
+            .map_err(|e| Error::ValidationError(e))?
+            .claims;
```

Comment:
It could be improved slightly.
```suggestion
            .map_err(Error::ValidationError)?
```

Reply:
fixed in [f45e131](https://github.com/JanssenProject/jans/pull/9726/commits/f45e1316be4b326597933d0b133bcc54aadadc2f)